extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/TransactionEntityManagers.java

```diff
@@ -39,8 +34,18 @@ public EntityManager getEntityManager(String unitName) {
             return entityManager;
         }
         return managers.computeIfAbsent(unitName, (un) -> new TransactionScopedEntityManager(
-                transactionManager, transactionSynchronizationRegistry, jpaConfig.getEntityManagerFactory(un), un,
+                getTransactionManager(), getTransactionSynchronizationRegistry(), jpaConfig.getEntityManagerFactory(un), un,
                 requestScopedEntityManagers));
     }
 
+    private TransactionManager getTransactionManager() {
```

Comment:
Don't you run the risk of those beans being removed by ArC's optimisation, or are they already unremovable if Agroal is present?

Reply:
Good question, I'm checking. Isn't it weird that no test failed?